Utils/Dataflow/pyDKB/dataflow/communication/stream/InputStream.py

```diff
@@ -16,6 +16,23 @@ class InputStream(Stream):
 
     __iterator = None
 
+    # Names of markers that the stream knows how to process.
+    # Values are taken from config, markers with empty values are ignored.
+    marker_names = ['eob']
+
+    def configure(self, config={}):
+        """ Configure instance. """
+        super(InputStream, self).configure(config)
+        self.markers = {}
+        # If batch size is 1, meaning non-batch mode will be used, then
+        # markers are unnecessary (and even can be a hindrance by forcing
+        # usage of custom_readline() without need).
+        if config.get('bsize', 1) > 1:
```

Comment:
Quick, non-conclusive comment:

> * as long as we're in the worker-driven model it makes no harm -- supervisor is not supposed to send `<EOB>` unless it received `<BNC>`, and with `batch_size == 1` the `<BNC>` won't ever be sent by the worker;

It will send `<BNC>` if the received message cannot be added to batch and thus prevents its size from increasing from 0 to 1. Message of wrong type, for example.

Reply:
So what's your idea?

Reply:
This was moved into email, resolving.